### PR TITLE
Discussions update

### LINKED ISSUE #613: If you navigate from a project with talk boards to one that doesn't using the project talk side bar then the old project's boards stay with new project's background
If you use the project Talk side bar to move to a project's talk without discussion boards, it leaves the previous projects boards in place 

see screenshots below:

I made this boards and threads for P4: Terrains:

screen shot 2015-06-19 at 6 05 51 am

Switched to Kitteh Zoo and the same P4: Terrains boards show

screen shot 2015-06-19 at 6 05 37 am

Same thing if I go to Snapshots at Sea

screen shot 2015-06-19 at 6 05 27 am

### PR DESCRIPTION
**Pull request by aweiksnar**
- Changes `current-section` to a function
  - Having this as a function works better than a mixin with Router#transitionTo, to ensure component is updated when props are received
  - Closes #613
- Also adds a `shouldComponentUpdate` to the project-linker, to prevent rerenders when changing between projects (looked a bit ugly with the loading state happening on every page change)

### CODE REVIEW

**Comment by chrissnyder** [comment]
Works as advertised. Couple comments inline.

File: app/talk/index.cjsx [hunk 1]
```diff
@@ -29,5 +29,5 @@ module?.exports = React.createClass
         </button>
       </form>
 
-      <RouteHandler {...@props} section={@state.currentSection} />
+      <RouteHandler {...@props} section={currentSection(@props.project)} />
```
Comment (chrissnyder) [inline app/talk/index.cjsx]: Couldn't this just be `<RouteHandler {...@props} section="zooniverse" />`? And then have the project-specific Talk instance use the `lib/project-section` function directly?

Comment (aweiksnar) [inline app/talk/index.cjsx]: :pensive: Yep, updated in c5e193d

**Comment by chrissnyder** [comment]
LGTM

**Pull request merged**